### DISCUSSION TITLE
Issue with Heroku Buildpack PHP

### DISCUSSION THREAD

**Discussion by dvd-km** [dokku/dokku#4562] [Q&A] [answered]
Hello Dokku community,

I’m trying to deploy some updates to our server and am coming up against the following error: 

remote: /app/.heroku/php-min/bin/php: error while loading shared libraries: libzip.so.4: cannot open shared object file: No such file or directory

I saw a github issue flagged on the [heroku-buildpack-php repo](https://github.com/heroku/heroku-buildpack-php/issues/387)  with the same error message and it looked answer given was to 'update the stack'. 

I found a bit more explanation from another issue [https://github.com/heroku/heroku-buildpack-php/issues/396#issuecomment-593646768](https://github.com/heroku/heroku-buildpack-php/issues/396#issuecomment-593646768) where they were saying that php 7.4 'now requires libonig for mbstring' and 'no longer bundles libzip, so it's needed on the system'. 

This is odd since we're not using php 7.4 ( we're using PHP 7.3.13 ) and any attempts to pin the heroku-buildpack-php to a specific version are failing on build too. 

I'm not sure how to go about 'updating the stack' to make it so that libzip is available for the system. ( Please forgive my cluelessness, I'm rather new to dokku and was not the person who initially set up this deploy flow. )

Has anyone here come across any issues similar to this and can share some insight on how I could go about addressing this issue? 

Much appreciated! 

Currently running on:

DigitialOcean Dokku Droplet
dokku version 0.21.4

ERROR LOG
-----> Adding BUILD_ENV to build environment...
-----> Warning: Multiple default buildpacks reported the ability to handle this app. The first buildpack in the list below will be used.
       Detected buildpacks: multi php
-----> Multipack app detected
=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php
=====> Detected Framework: PHP
remote: -----> Bootstrapping...
remote: /app/.heroku/php-min/bin/php: error while loading shared libraries: libzip.so.4: cannot open shared object file: No such file or directory

**Comment by josegonzalez** [comment, marked as answer]
You should update both dokku and herokuish via apt:

```
apt-get update
apt-get install dokku herokuish
```

Herokuish is equivalent to a "stack" in Heroku's terminology.
